examples/2-rollout-setHeaderRoute/rollout.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: gloo-rollouts-demo
+  namespace: gloo-rollouts-demo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: gloo-rollouts-demo
+  template:
+    metadata:
+      labels:
+        app: gloo-rollouts-demo
+    spec:
+      containers:
+      - image:  argoproj/rollouts-demo:blue
+        imagePullPolicy: IfNotPresent
+        name: gloo-rollouts-demo
+        ports:
+        - containerPort: 8080
+        resources:
+          limits:
+            cpu: 100m
+            memory: 100Mi
+  strategy:
+    canary:
+      canaryService: canary
+      stableService: stable
+      trafficRouting:
+        # managedRoutes are required for using setHeaderRoute
+        # routes specified here are owned by the plugin
+        # the plugin will create/delete these routes as needed
+        # do not specify routes which are already otherwise in use under this field
+        managedRoutes: 
+          - name: header-canary
```

Comment:
It'd be great if we could detect if a Route Table has never been affected by a Rollout so that we can error. Not quite sure something like that exists.

Reply:
Off the top of my head the best I can think of would be to warn that a managed route with the name we are meant to add was found while we're adding the route